src/controllers/connection/newConnection.ts

```diff
@@ -44,30 +48,41 @@ export class NewConnectionController extends HTMLController {
   @SuccessResponse(200)
   @Get('/verify-company')
   public async verifyCompanyForm(@Query() companyNumber: string): Promise<HTML> {
-    this.logger.debug('connections page requested')
     const regex =
       /^(((AC|CE|CS|FC|FE|GE|GS|IC|LP|NC|NF|NI|NL|NO|NP|OC|OE|PC|R0|RC|SA|SC|SE|SF|SG|SI|SL|SO|SR|SZ|ZC|\d{2})\d{6})|((IP|SP|RS)[A-Z\d]{6})|(SL\d{5}[\dA]))$/
     if (!regex.test(`${companyNumber}`)) {
       return this.html(
-        this.newConnection.companyNumberInput({
-          targetBox: await this.newConnection.companyEmptyTextBox({ errorMessage: 'Company number format incorrect' }),
+        this.newConnection.companyFormInput({
+          targetBox: {
+            status: 'error',
+            errorMessage: 'Company number format incorrect',
+          },
+          formStage: 'form',
         })
       )
     }
-    let company
+    let company: CompanyProfile
```

Comment:
wonder if this could be added as a function variable with a default value like `null`

so function would look like

```js
public async verifyCompanyForm(@Query() companyNumber: string, company: CompanyProfile = undefined): Promise<HTML> {
  ...
}
```

This way we could avoid using mutable `let`

Reply:
we can't put this in as a parameter (we need to derive the profile using the company number from companies house) but you can look to refactor things so we avoid let